### PR TITLE
refactor get and process functionality of queue

### PR DESCRIPTION
**Pull request by ramaraochavali**
To better understand and segregate get and process functions of queue

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pkg/queue/instance.go [hunk 1]
```diff
@@ -117,6 +117,48 @@ func (q *queueImpl) Closed() <-chan struct{} {
...
+// the processing go routine should stop.
+func (q *queueImpl) get() (task *BackoffTask, shutdown bool) {
+	q.cond.L.Lock()
+	defer q.cond.L.Unlock()
+	// wait for closing to be set, or a task to be pushed
+	for !q.closing && len(q.tasks) == 0 {
+		q.cond.Wait()
+	}
+
+	if len(q.tasks) == 0 {
```
Comment (hzxuzhonghu) [inline pkg/queue/instance.go:130]: This is not equal to previous behavior

Comment (ramaraochavali) [reply pkg/queue/instance.go:130]: I can change to previous condition but it is same behaviour. Wait only comes out if len(q.tasks) is zero

Comment (hzxuzhonghu) [reply pkg/queue/instance.go]: Previously, if queue is stopped, no tasks will be run if there is any left unprocessed.

Comment (ramaraochavali) [reply pkg/queue/instance.go]: ok. any way I changed to previous condition.

**Review by hzxuzhonghu — COMMENTED** [review]
LGTM, but need more eyes on it.

**Review by howardjohn — APPROVED** [review]
LGTM, seems just a refactor with no behavior change

**Pull request merged**